Sources/EasyPurchase/PaymentQueueController.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  PaymentQueueController.swift
+//  EasyPurchase
+//
+//  Created by Sadat Ahmed on 20.09.2023.
+//
+
+import Foundation
+import StoreKit
+
+// Protocol for a payment queue
+protocol PaymentQueue {
+    func add(_ payment: SKPayment)
+}
+
+// Define the PaymentQueueDelegate protocol for handling payment queue events
+protocol PaymentQueueDelegate: AnyObject {
+    func handleTransactionSuccess(for productIdentifier: String)
+    func handleTransactionFailure(for productIdentifier: String, with error: Error?)
+    func processTransactions(_ transactions: [SKPaymentTransaction], on paymentQueue: PaymentQueue) -> [SKPaymentTransaction]
+}
+
+// PaymentQueueController class responsible for managing the payment queue
+class PaymentQueueController: NSObject, PaymentQueue, SKPaymentTransactionObserver {
```

Comment:
if this class needs to restrict future inheritance then it should be set as final class